### ISSUE TITLE
[DocDB] TableLocks: Handle abort of a deadlocked virtual transaction

### ISSUE THREAD

**Issue by basavaraj29** [yugabyte/yugabyte-db#24628]
Jira Link: [DB-13685](https://yugabyte.atlassian.net/browse/DB-13685)
### Description

Once https://github.com/yugabyte/yugabyte-db/issues/24596 lands, we could run into scenarios where the deadlock cycle contains virtual txns alone and no regular distributed txn (with object locks / advisory locks enabled).

We need to handle the abort path for a virtual transaction. We should have a mechanism to unblock the waiting requests at the object lock manager (as there is no regular txn, all the waiting requests would have been at the OLM itself), but not abort the virtual txn itself (since that corresponds to the host pg session).

### Issue Type

kind/new-feature

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-13685]: https://yugabyte.atlassian.net/browse/DB-13685?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by basavaraj29** [comment]
No longer relevant as we dropped the idea of a virtual transaction all together.
